flask/airbrake_middleware.py

```diff
@@ -0,0 +1,46 @@
+import sys
+import datetime
+from flask import g, request
+from argparse import ArgumentParser
+
+sys.path.append('../pybrake')
+import routes
+from utils import logger
+
+parser = ArgumentParser()
+parser.add_argument("-project_id", dest="project_id", help="airbrake project ID")
+parser.add_argument("-project_key", dest="project_key", help="airbrake project key")
+parser.add_argument("-host", dest="host", help="airbrake host")
+
+args = parser.parse_args()
+routeStats = routes.RouteStats(args.project_id, args.project_key, args.host)
+
+def setup_airbrake_middleware(app):
+  app.before_request(before_request)
+  app.after_request(after_request)
+
+def before_request():
+  g.flask_request_start_time = datetime.datetime.now()
+
+def after_request(response):
+  if not hasattr(g, 'flask_request_start_time'):
+    logger.error("flask_request_start_time is empty")
+    return response
+
+  now = datetime.datetime.utcnow()
+  dur = now - g.flask_request_start_time
+  if dur == 0:
+    logger.error("processing time is not valid")
+    return response
+
+  stat = {
+    "method": request.method,
+    "route": str(request.endpoint),
+    "status_code": response.status_code,
+    "time": now.replace(second=0, microsecond=0).strftime('%Y-%m-%dT%H:%M:%SZ'),
+    "dur": dur.microseconds,
+  }
+  routeStats.inc_request(stat)
```

Comment:
this should be `routeStats.inc_request(method=request.method, route=str(request.endpoint), etc)`